File: Data_Structures/Queues/queue_array.c
// A program to implement a queue using an array and perform basic operations like enqueue and dequeue

#include <stdio.h>
#define MAX_SIZE 100

int queue[MAX_SIZE];
int front = -1, rear = -1;

void enqueue(int value) {
    if (rear == MAX_SIZE - 1) {
        printf("Queue is full. Cannot enqueue.\n");
        return;
    }
    if (front == -1)
        front = 0;
    rear++;
    queue[rear] = value;
}

void dequeue() {
    if (front == -1 || front > rear) {
        printf("Queue is empty. Cannot dequeue.\n");
        return;
    }
    printf("Dequeued element: %d\n", queue[front]);
    front++;
}

int main() {
    enqueue(5);
    enqueue(10);
    enqueue(15);
    dequeue();
    enqueue(20);
    dequeue();
    dequeue();

    printf("\n");
    return 0;
}

File: Data_Structures/Queues/queue_array.py
# A program to implement a queue using an array and perform basic operations like enqueue and dequeue

class Queue:
    def __init__(self):
        self.queue = []

    def enqueue(self, value):
        self.queue.append(value)

    def dequeue(self):
        if not self.queue:
            print("Queue is empty. Cannot dequeue.")
            return
        print("Dequeued element:", self.queue.pop(0))

queue = Queue()
queue.enqueue(5)
queue.enqueue(10)
queue.enqueue(15)
queue.dequeue()
queue.enqueue(20)
queue.dequeue()
queue.dequeue()
queue.dequeue()
queue.dequeue()